cpp/include/cudf/io/parquet.hpp

```diff
@@ -0,0 +1,847 @@
+   */
+  statistics_freq get_stats_level() const { return _stats_level; }
+
+  /**
+   * @brief Returns nullable metadata information
+   */
+  const table_metadata_with_nullability* get_nullable_metadata() const
+  {
+    return _nullable_metadata;
+  }
+
+  /**
+   * @brief Set nullable metadata to parquet_writer_options
+   *
+   * @param metadata associated metadata
+   */
+  void set_nullable_metadata(const table_metadata_with_nullability* metadata)
+  {
+    _nullable_metadata = metadata;
+  }
+
+  /**
+   * @brief Set statistics_freq to parquet_writer_options
+   *
+   * @param sf statistics_freq that needs to be written to output
+   */
+  void set_stats_level(statistics_freq sf) { _stats_level = sf; }
+
+  /**
+   * @brief Set compression type to parquet_writer_options
+   *
+   * @param comp_type compression_type to use
+   */
+  void set_compression(compression_type comp_type) { _compression = comp_type; }
+
+  /**
+   * @brief creates builder to build chunked_parquet_writer_options.
+   *
+   * @param sink sink to use for writer output
+   *
+   * @return chunked_parquet_writer_options_builder
+   */
+  static chunked_parquet_writer_options_builder builder(sink_info const& sink);
+};
+
+class chunked_parquet_writer_options_builder {
+  chunked_parquet_writer_options options;
+
+ public:
+  // This is just to please cython
+  /**
+   * @brief Default constructor
+   */
+  chunked_parquet_writer_options_builder() = default;
+
+  /**
+   * @brief Constructor from sink.
+   *
+   * @param sink sink to use for writer output
+   */
+  chunked_parquet_writer_options_builder(sink_info const& sink) : options(sink){};
+
+  /**
+   * @brief Set nullable metadata to parquet_writer_options
+   *
+   * @param metadata associated metadata
+   * @return reference to chunked_parquet_writer_options_builder
+   */
+  chunked_parquet_writer_options_builder& set_nullable_metadata(
+    const table_metadata_with_nullability* metadata)
+  {
+    options._nullable_metadata = metadata;
+    return *this;
+  }
+
+  /**
+   * @brief Set statistics_freq to parquet_writer_options
+   *
+   * @param sf statistics_freq that needs to be written to output
+   * @return reference to chunked_parquet_writer_options_builder
+   */
+  chunked_parquet_writer_options_builder& set_stats_level(statistics_freq sf)
+  {
+    options._stats_level = sf;
+    return *this;
+  }
+
+  /**
+   * @brief Set compression type to parquet_writer_options
+   *
+   * comp_type compression_type to use
+   * @return reference to chunked_parquet_writer_options_builder
+   */
+  chunked_parquet_writer_options_builder& set_compression(compression_type comp_type)
+  {
+    options._compression = comp_type;
+    return *this;
+  }
+
+  /**
+   * @brief move chunked_parquet_writer_options member once options is built
+   */
+  operator chunked_parquet_writer_options &&() { return std::move(options); }
+
+  /**
+   * @brief move chunked_parquet_writer_options member once options is built
```

Comment:
```suggestion
   * @brief move chunked_parquet_writer_options member once it's built
   *
   * This has been added since Cython does not support overloading of conversion operators.
```